core/src/main/java/org/apache/carbondata/core/util/TrashUtil.java

```diff
@@ -268,4 +268,45 @@ public static String getCompleteTrashFolderPath(String tablePath, long timeStamp
       timeStampSubFolder + CarbonCommonConstants.FILE_SEPARATOR + CarbonTablePath
       .SEGMENT_PREFIX + segmentNumber;
   }
+
+  /**
+   * This will give the complete path of the trash folder with the timestamp and the partition name
+   *
+   * @param tablePath          absolute table path
+   * @param timeStampSubFolder the timestamp for the clean files operation
+   * @param partitionName      partition name for which files are moved to the trash folder
+   */
+  public static String getCompleteTrashFolderPathForPartition(String tablePath,
+      long timeStampSubFolder, String partitionName) {
+    return CarbonTablePath.getTrashFolderPath(tablePath) + CarbonCommonConstants.FILE_SEPARATOR
+        + timeStampSubFolder + CarbonCommonConstants.FILE_SEPARATOR
+        + partitionName;
+  }
+
+  /**
+   * The below method copies dropped partition files to the trash folder.
+   *
+   * @param filesToCopy              absolute path of the files to copy to the trash folder
+   * @param trashFolderWithTimestamp trashfolderpath with complete timestamp and segment number
+   */
+  public static void copyFilesToTrash(String filesToCopy, String trashFolderWithTimestamp) {
```

Comment:
why are different methods required for handling partition tables? the existing method does not handle it? if so change the method name specific to the partition table, so that no one reuses it by mistake

Reply:
yes, method name changed accordingly